src/routes/SafeAppLandingPage/components/UserSafeSection.tsx

```diff
@@ -0,0 +1,141 @@
+import { ReactElement } from 'react'
+import { useSelector } from 'react-redux'
+import { Title } from '@gnosis.pm/safe-react-components'
+import styled from 'styled-components'
+
+import { userAccountSelector } from 'src/logic/wallets/store/selectors'
+import ConnectButton from 'src/components/ConnectButton'
+import { AddressBookEntry } from 'src/logic/addressBook/model/addressBook'
+import useOwnerSafes from 'src/logic/safe/hooks/useOwnerSafes'
+import useLocalSafes, { LocalSafes } from 'src/logic/safe/hooks/useLocalSafes'
+import { lastViewedSafe } from 'src/logic/currentSession/store/selectors'
+import { addressBookState } from 'src/logic/addressBook/store/selectors'
+import CreateNewSafe from './CreateNewSafe'
+import UseYourSafe from './UseYourSafe'
+
+type UserSafeProps = {
+  safeAppUrl: string
+  availableChains: string[]
+  safeAppChainId: string | null
+}
+
+const UserSafeSection = ({ safeAppUrl, availableChains, safeAppChainId }: UserSafeProps): ReactElement => {
+  const userAddress = useSelector(userAccountSelector)
+  const lastViewedSafeAddress = useSelector(lastViewedSafe)
+  const ownedSafes = useOwnerSafes()
+  const localSafes = useLocalSafes()
+  const addressBook = useSelector(addressBookState)
+
+  const compatibleUserSafes = getCompatibleSafes(ownedSafes, localSafes, availableChains, safeAppChainId, addressBook)
+
+  const selectedUserSafe = getDefaultSafe(compatibleUserSafes, lastViewedSafeAddress, safeAppChainId)
+
+  const isWalletConnected = !!userAddress
+
+  return (
+    <UserSafeContainer>
+      <Title size="xs">Use the dApp with your Safe!</Title>
+      {isWalletConnected ? (
+        selectedUserSafe ? (
+          <UseYourSafe safeAppUrl={safeAppUrl} defaultSafe={selectedUserSafe} safes={compatibleUserSafes} />
+        ) : (
+          <CreateNewSafe safeAppUrl={safeAppUrl} />
+        )
+      ) : (
+        <ConnectWalletContainer>
+          <ConnectWalletButton data-testid="connect-wallet-btn" />
+        </ConnectWalletContainer>
+      )}
+    </UserSafeContainer>
+  )
+}
+
+export default UserSafeSection
+
+const UserSafeContainer = styled.div`
+  flex: 1 0 50%;
+
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`
+
+const ConnectWalletContainer = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  height: 100%;
+`
+const ConnectWalletButton = styled(ConnectButton)`
+  height: 52px;
+`
+
+const getCompatibleSafes = (
```

Comment:
Perhaps some well named functions instead anonymous (filters, maps, reducers) here can help to follow the code

Reply:
Even creating a module with all the stuff